sphinx_qt_documentation/_missing_reference.py

```diff
@@ -0,0 +1,226 @@
+
+from docutils import nodes
+from docutils.nodes import Element, TextElement
+from sphinx.application import Sphinx
+from sphinx.environment import BuildEnvironment
+from sphinx.ext.intersphinx import InventoryAdapter
+from sphinx.locale import get_translation
+
+_ = get_translation("sphinx")
+
+
+def _get_signal_and_version():
+    name_mapping = {
+        "qtpy": (lambda: "QT_VERSION", "Signal"),
+        "Qt": (lambda: "__qt_version__", "Signal"),
+        "PySide2": (
+            lambda: importlib.import_module("PySide2.QtCore").__version__,
+            "Signal",
+        ),
+        "PyQt5": (
+            lambda: importlib.import_module("PyQt5.QtCore").QT_VERSION_STR,
+            "pyqtSignal",
+        ),
+    }
+    for module_name, (version, signal_name) in name_mapping.items():
+        try:
+            core = importlib.import_module(f"{module_name}.QtCore")
+            signal = getattr(core, signal_name)
+            return signal, version()  # pylint: disable=E1102
+        except ImportError:
+            continue
+    raise RuntimeError("No Qt bindings found")
+
+
+Signal, QT_VERSION = _get_signal_and_version()
+
+signal_slot_uri = {
+    "Qt5": "https://doc.qt.io/qt-5/signalsandslots.html",
+    "PySide2": "https://doc.qt.io/qtforpython/overviews/signalsandslots.html",
+    "PyQt5": "https://www.riverbankcomputing.com/static/Docs/PyQt5/signals_slots.html",
+}
+signal_name_dict = {"Qt5": "Signal", "PySide2": "Signal", "PyQt5": "pyqtSignal"}
+slot_name = {"Qt5": "Slot", "PySide2": "Slot", "PyQt5": "pyqtSlot"}
+type_translate_dict = {
+    "class": ["class", "enum", "attribute"],
+    "meth": ["method", "signal"],
+    "mod": ["module"],
+}
+signal_pattern = re.compile(r"((\w+\d?\.QtCore\.)|(QtCore\.)|(\.))?(pyqt)?Signal")
+slot_pattern = re.compile(r"((\w+\d?\.QtCore\.)|(QtCore\.)|(\.))?(pyqt)?Slot")
+
+
+def _fix_target(target: str, inventories: InventoryAdapter) -> str:
+    if (
+        target.startswith("PySide2")
+        and "PySide2" not in inventories.named_inventory
+        and "PyQt5" in inventories.named_inventory
+    ):
+        _head, dot, tail = target.partition(".")
+        return "PyQt5" + dot + tail
+    if (
+        target.startswith("PyQt5")
+        and "PyQt5" not in inventories.named_inventory
+        and "PySide2" in inventories.named_inventory
+    ):
+        _head, dot, tail = target.partition(".")
+        return "PySide2" + dot + tail
+    return target
+
+
+def _get_inventory_for_target(target: str, inventories: InventoryAdapter):
+    try:
+        return inventories.named_inventory[target.partition(".")[0]]
+    except KeyError:
+        pass
+    if "Qt" in inventories.named_inventory:
+        return inventories.named_inventory["Qt"]
+    if "Qt5" in inventories.named_inventory:
+        return inventories.named_inventory["Qt5"]
+    if "PyQt5" in inventories.named_inventory:
+        return inventories.named_inventory["PyQt5"]
+    return inventories.named_inventory["PySide2"]
```

Comment:
@altendky here is the code for select which inventory is searched. Did you have any idea how to check if it is a real Qt inventory?

Reply:
The first check using `.partition()` will get a non-Qt inventory such as `trio` in the example error case of looking up `trio.abc.Clock`.  So, this doesn't play well with our search technique of trying lots of prefixes below in `_extract_from_inventory()` whereby we effectively change the first segment that would be used here.

Reply:
I guess there's some question of priority.  Usually it won't matter, but let's say the target is `blue.red` and there is both a `blue` inventory and a `PySide2.blue.red`.  Do we ignore the item and let whatever else handle it via the `blue` inventory or do we claim it as `PySide2.blue.red`?  I think deciding that may help with either "giving up" here (if we want to let something else handle it) or go attempt to find it later and only give up when we can't find a solution (if we want to claim it ourselves).

Reply:
So maybe introduce a close list of accepted prefixes?

Reply:
I do think we should not be searching inventories other than what we know how to handle.  It would be nice to not have to track a list of all the Qt modules like `["QtCore", "QtGui", "QtWidgets", ...]`.